inference/core/models/instance_segmentation_base.py

```diff
@@ -240,14 +242,19 @@ def make_response(
             - For each image, constructs an `InstanceSegmentationInferenceResponse` object.
             - Each response contains a list of `InstanceSegmentationPrediction` objects.
         """
-        responses = [
-            InstanceSegmentationInferenceResponse(
-                predictions=[
+        responses = []
+        for ind, (batch_predictions, batch_masks) in enumerate(zip(predictions, masks)):
+            predictions = []
+            for pred, mask in zip(batch_predictions, batch_masks):
+                if class_filter and self.class_names[int(pred[6])] in class_filter:
+                    # TODO: logger.debug
+                    continue
+                # Passing args as a dictionary here since one of the args is 'class' (a protected term in Python)
+                predictions.append(
                     InstanceSegmentationPrediction(
-                        # Passing args as a dictionary here since one of the args is 'class' (a protected term in Python)
                         **{
-                            "x": (pred[0] + pred[2]) / 2,
-                            "y": (pred[1] + pred[3]) / 2,
+                            "x": pred[0] + (pred[2] - pred[0]) / 2,
```

Comment:
I understand that this change is to make it verbose that we are adding half width / height in order to make x, y point center of bbox, right?
in other words - there is no math diff (despite potential floating point diffs)

Reply:
Yep, I know the math before was correct but thought my brain will not think about these lines when I read this again. If this results in different result due to floating point error it's not surfacing int ests.